Source/Editor/Surface/IParametersDependantNode.cs

```diff
@@ -33,5 +33,13 @@ public interface IParametersDependantNode
         /// </summary>
         /// <param name="param">The parameter.</param>
         void OnParamDeleted(SurfaceParameter param);
+
+        /// <summary>
+        /// Get if the parameter is referenced in a graph. Referenced in this case means in a graph and at least one node in-/output connected to another node.
+        /// </summary>
+        /// <param name="param">The parameter.</param>
+        /// <param name="surface">The visject surface.</param>
+        /// <returns>If the parameter is referenced.</returns>
+        bool IsParamreferenced(SurfaceParameter param, VisjectSurface surface);
```

Comment:
Drop `VisjectSurface surface` parameter as node itself knows about internals and can detect whether uses that param.

Reply:
Added in https://github.com/FlaxEngine/FlaxEngine/pull/3010/commits/b055117144d42df15435a98207e41299161185c6